feat(dp): Implement GetState in DP orc8r service

WojSad opened the pull request:
## Summary

As part of Radio Controller's migration to orc8r Implemented GetState method for the new AmcManager interface in golang.

## Test Plan

Tests written in go to reflect current tests in python

jkmar commented on dp/cloud/go/services/dp/storage/db/finishing.go:
```diff
@@ -93,6 +93,8 @@ func (q *Query) List() ([][]Model, error) {
 	c := collectColumns(q)
 	query := buildQuery(q.builder, q, q.arg.filter, c.getColumnNames())
 	query = addPagination(q.pagination, query)
+	s, _, _ := query.ToSql()
```
Please remove this debug print

jkmar commented on dp/cloud/go/services/dp/storage/dp_manager.go:
```diff
@@ -0,0 +1,35 @@
+	sq "github.com/Masterminds/squirrel"
+)
+
+type DpSqlManager interface {
```
I think we don't need to define this interface as it is unused

WojSad replied:
done

jkmar commented on dp/cloud/go/services/dp/storage/amc_manager_test.go:
```diff
@@ -0,0 +1,695 @@
+	}}
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			for _, c := range tc.inputCbsds {
```
You can do this to keep setup in one transaction:
```
testCases := []struct {
		name          string
		input []db.Model // keep cbsds grants, requests here
		expected      []*storage.DetailedCbsd
	}{{
````

WojSad replied:
done

jkmar commented on dp/cloud/go/services/dp/storage/amc_manager_test.go:
```diff
@@ -0,0 +1,695 @@
+				s.givenResourcesInserted(g)
+			}
+
+			// actual, err := sqorc.ExecInTx(s.database, nil, nil, func(tx *sql.Tx) (interface{}, error) {
```
Please remove commented code

WojSad replied:
done

jkmar commented on dp/cloud/go/services/dp/storage/amc_manager_test.go:
```diff
@@ -0,0 +1,652 @@
+	}}
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			for _, m := range tc.input {
```
Actually now you can just run
```
s.resourceManager.InsertResources(db.NewExcludeMask(), tc.input...)
```
so that everything is in one transaction

jkmar commented on dp/cloud/go/services/dp/storage/amc_manager_test.go:
```diff
@@ -0,0 +1,652 @@
+		input: []db.Model{
+			b.NewDBCbsdBuilder().
+				WithNetworkId(someNetwork).
+				WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 14)).
```
I would do this `fmt.Sprintf("%s%d", someSerialNumber, 14)` or this `someSerialNumber+"14"`, to avoid mixing different methods of building strings

Commit:
feat(dp): Implement GetState in DP orc8r service

As part of Radio Controller's migration to orc8r Implemented GetState method for the new AmcManager interface in golang

Signed-off-by: Wojciech Sadowy <wojciech.sadowy@freedomfi.com>
